File: ruby/func.rb
if __FILE__ == $0
	require './func'
	print "Please enter an integer value: "
	n = gets.chomp.to_i
	r = fact(n)
	puts "Fact(" + n.to_s + ")= " + r.to_s
end

def fact(i)
	if i <= 1 
		1
	else
		i * fact(i-1)
	end
end

File: cpp/func.cpp
#include <iostream>
int fact(int i);
int main()
{
  int n;
  std::cout << "Please enter an integer value: ";
  std::cin >> n;
  std::cout << "Fact(" << n << ")= ";
  std::cout << fact(n) << "\n";
}

int fact(int i){
	if (i <=1)
		return 1;
	return i * fact(i-1);
}
